File: CS/CustomFunctionForExpressionEditorExample/Functions/Component1.cs
using System.ComponentModel;

namespace CustomFunctionForExpressionEditorExample.Functions
{
    public partial class Component1 : Component
    {
        public Component1()
        {
            InitializeComponent();
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(new NewLineConstant());
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(new CustomFormatFunction());
        }

        public Component1(IContainer container)
        {
            container.Add(this);

            InitializeComponent();
        }
    }
}

File: VB/CustomFunctionForExpressionEditorExample/Functions/Component1.vb
Imports CustomFunctionForExpressionEditorExample

Public Class Component1
    Inherits System.ComponentModel.Component
    Public Sub New()
        DevExpress.XtraReports.Expressions.CustomFunctions.Register(New NewLineConstant())
        DevExpress.XtraReports.Expressions.CustomFunctions.Register(New CustomFormatFunction())
    End Sub
End Class
